core/controllers/blog_admin.py

```diff
@@ -0,0 +1,180 @@
+            },
+        }
+    }
+
+    @acl_decorators.can_access_blog_admin_page
+    def get(self):
+        # type: () -> None
+        """Handles GET requests."""
+        config_properties = config_domain.Registry.get_config_property_schemas()
+        config_prop_for_blog_admin = {
+            'list_of_default_tags_for_blog_post': (
+                config_properties['list_of_default_tags_for_blog_post']),
+            'max_number_of_tags_assigned_to_blog_post': (
+                config_properties['max_number_of_tags_assigned_to_blog_post'])
+        }
+        role_to_action = role_services.get_role_actions()
+        self.render_json({
+            'config_properties': config_prop_for_blog_admin,
+            'role_to_actions': {
+                BLOG_POST_EDITOR: role_to_action[BLOG_POST_EDITOR],
+                BLOG_ADMIN: role_to_action[BLOG_ADMIN]
+            },
+            'updatable_roles': {
+                BLOG_POST_EDITOR: (
+                    role_services.HUMAN_READABLE_ROLES[BLOG_POST_EDITOR]),
+                BLOG_ADMIN: role_services.HUMAN_READABLE_ROLES[BLOG_ADMIN]
+            }
+        })
+
+    @acl_decorators.can_access_blog_admin_page
+    def post(self):
+        # type: () -> None
+        """Handles POST requests."""
+        try:
+            result = {}
+            if self.normalized_payload.get(
+                    'action') == 'save_config_properties':
+                new_config_property_values = self.normalized_payload.get(
+                    'new_config_property_values')
+                logging.info(
+                    '[BLOG ADMIN] %s saved config property values: %s' %
+                    (self.user_id, new_config_property_values))
+                for (name, value) in new_config_property_values.items():
+                    config_services.set_property(self.user_id, name, value)
+            elif self.normalized_payload.get(
+                    'action') == 'revert_config_property':
+                config_property_id = (
+                    self.normalized_payload.get('config_property_id'))
+                config_services.revert_property(
+                    self.user_id, config_property_id)
+                logging.info(
+                    '[BLOG ADMIN] %s reverted config property: %s' %
+                    (self.user_id, config_property_id))
+            self.render_json(result)
+        except Exception as e:
+            logging.exception('[BLOG ADMIN] %s', e)
+            self.render_json({'error': python_utils.UNICODE(e)})
+            python_utils.reraise_exception()
+
+
+class BlogAdminRolesHandler(base.BaseHandler):
+    """Handler for the blog admin page."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'POST': {
+            'action': {
+                'type': 'basestring',
+            },
+            'role': {
+                'type': 'basestring'
```

Comment:
Please add the choices(BLOG_ADMIN, BLOG_POST_EDITOR) and remove the invalid input exception.

Reply:
Done